src/MPAGSCipher/CaesarCipher.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef CAESARCIPHER_HPP
+#define CAESARCIPHER_HPP
+
+#include <string>
+#include <vector>
+#include <CipherMode.hpp>
+
+class CaesarCipher{
+    public:
+        explicit CaesarCipher(std::size_t key);
+        explicit CaesarCipher(std::string key);
+        std::string applyCipher(std::string inputText, CipherMode mode);
```

Comment:
Suggest a few additions of `const` and `std::string` objects should probably be taken by reference for efficiency:
```suggestion
        explicit CaesarCipher(const std::size_t key);
        explicit CaesarCipher(const std::string& key);
        std::string applyCipher(const std::string& inputText, const CipherMode mode);
```